fuels-contract/src/contract.rs

```diff
@@ -309,12 +335,23 @@ pub struct ContractCall<D> {
     pub datatype: PhantomData<D>,
     pub output_params: Vec<ParamType>,
     pub custom_inputs: bool,
+    external_contracts: Option<Vec<ContractId>>,
 }
 
 impl<D> ContractCall<D>
 where
     D: Detokenize,
 {
+    /// Sets external contracts as dependencies to this contract's call.
+    /// Effectively, this will be used to create Input::Contract/Output::Contract
+    /// pairs and set them into the transaction.
+    /// Note that this is a builder method, i.e. use it as a chain:
+    /// `my_contract_instance.my_method(...).set_contracts(&[another_contract_id]).call()`.
```

Comment:
That's a good thought that would be very cool from a UX perspective. But unfortunately no, it's not desirable. Setting an external contract as a dependency to a contract method call is something that happens within a single transaction, therefore this happens _per call_. This effectively updates the `Transaction` with `Input::Contract`s and `Output::Contract`s. 

For instance, there are cases where in a call/transaction we use contract ID `A`, and in another call, we use a contract ID `B`[0]. If you had set the contract ID at the instantiation level (when creating the contract instance), this wouldn't be possible to achieve.  

[0]: You could imagine a `transfer` method that would use `n` underlying contracts to perform the transfer. In a first call you would like to `contract.transfer(10).set_contracts(vec![A])` so it would use `A`, and in another call you would like to use the contract `B` for that: `contract.transfer(10).set_contracts(vec![B])`.

TL;DR: these Input/Output contracts shouldn't be bound to the contract instance, it should be bound to a transaction instance.

Reply:
Ok got it, thanks for explaining.